scripts/bin/revbump

```diff
@@ -1,95 +1,89 @@
-		_buildsh_path="${REPO_ROOT}/${repo_path}/${package}/build.sh"
-		repo=$(jq --raw-output ".[\"${repo_path}\"].repo" ${REPO_ROOT}/repo.json)
-		echo $_buildsh_path
-
-		if [ -f "${_buildsh_path}" ]; then
-			buildsh_path=$_buildsh_path
-			break
-		fi
-	done
-
-	if [ -z "$buildsh_path" ]; then
-		echo "${package}: skipping as no build.sh found"
-		continue
-	fi
-
-	if grep -qP '^TERMUX_PKG_REVISION=(\d+)$' "${buildsh_path}"; then
-		# Increment revision.
-		current_rev=$(. "${buildsh_path}" 2>/dev/null; echo "${TERMUX_PKG_REVISION}")
-		incremented_rev=$((current_rev + 1))
-
-		# Replace revision value in build.sh.
-		if ! ${IS_GIT_REPOSITORY}; then
-			echo "${package}: bumping to ${incremented_rev}"
-		fi
-		sed -i -E "s/^(TERMUX_PKG_REVISION=)(.*)$/\1${incremented_rev}/g" "${buildsh_path}"
-	else
-		# Add base (1) revision right after TERMUX_PKG_VERSION.
-		if ! ${IS_GIT_REPOSITORY}; then
-			echo "${package}: setting revision to 1"
-		fi
-		sed -i -E "/TERMUX_PKG_VERSION=(.*)/a TERMUX_PKG_REVISION=1" "${buildsh_path}"
-	fi
-
-	# If we are on Git repository, prompt for committing changes.
-	if ${IS_GIT_REPOSITORY}; then
-		echo
-		echo "You are about to commit these changes:"
-		echo
-		echo "--------------------"
-		git --no-pager diff --patch "${buildsh_path}"
-		echo "--------------------"
-		echo
-		echo "rebuild(${repo}/${package}): $1"
-		echo
-		read -re -p "Do you want to commit these changes ? (y/n) " CHOICE
-		echo
-		if [[ ${CHOICE} =~ (Y|y) ]]; then
-			git add "${buildsh_path}"
-			git commit -m "rebuild(${repo}/${package}): $1"
-		else
-			echo "Not committing to Git!"
-		fi
-		echo
-	fi
-done
+#!/usr/bin/env python3
+import os
+import sys
+
+filename = sys.argv[0].split("/")[-1]
+
+usage = f"""
+Usage:
+
+{filename} <package> ...            
+
+Add or increment TERMUX_PKG_REVISION of package. 
+
+
+{filename} --dependencies  <package>    
+
+Add or increment TERMUX_PKG_REVISION of all the packages that have <package> in TERMUX_PKG_DEPENDS or in TERMUX_PKG_BUILD_DEPENDS
+"""
+
+
+FOLDERS = ["packages","x11-packages","root-packages"]
+
+
+def get_build_dependent_files(folders: list, DEP : str) -> list:
+    build_files = []
+    for i in folders:
+        for folder in os.listdir(i):
+            for line in open(i+"/"+folder+"/build.sh","r").read().split("\n"):
+                if line.startswith("TERMUX_PKG_DEPENDS") or line.startswith("TERMUX_PKG_BUILD_DEPENDS"):
+                    if DEP in line:
+                        build_files.append(i+"/"+folder+"/build.sh")
+    return build_files
+
+
+def bump_revision(file:str) -> None:
+    """ 
+    Bumps version in file
+    """
+    file_read = open(file,"r").read()
+    revision = 0
+    data = ""
+
+    if "TERMUX_PKG_REVISION" in file_read:
+        for line in file_read.split("\n"):
+            if line.startswith("TERMUX_PKG_REVISION"):
+                revision = int(line.split("=")[-1])+1
+                data = file_read.split("\n")
```

Comment:
This, and the following lines, do not have to be in the inner loop. Also seems easier to just `split("\n")` once before the loop, and then iterate over the list of lines